crates/scroll/chainspec/src/lib.rs

```diff
@@ -0,0 +1,605 @@
+    }
+
+    /// Build the resulting [`ScrollChainSpec`].
+    ///
+    /// # Panics
+    ///
+    /// This function panics if the chain ID and genesis is not set ([`Self::chain`] and
+    /// [`Self::genesis`])
+    pub fn build(self) -> ScrollChainSpec {
+        ScrollChainSpec { inner: self.inner.build() }
+    }
+}
+
+/// Scroll chain spec type.
+#[derive(Debug, Clone, Deref, Into, Constructor, PartialEq, Eq)]
+pub struct ScrollChainSpec {
+    /// [`ChainSpec`].
+    pub inner: ChainSpec,
+}
+
+impl EthChainSpec for ScrollChainSpec {
+    fn chain(&self) -> alloy_chains::Chain {
+        self.inner.chain()
+    }
+
+    fn base_fee_params_at_block(&self, block_number: u64) -> BaseFeeParams {
+        self.inner.base_fee_params_at_block(block_number)
+    }
+
+    fn base_fee_params_at_timestamp(&self, timestamp: u64) -> BaseFeeParams {
+        self.inner.base_fee_params_at_timestamp(timestamp)
+    }
+
+    fn deposit_contract(&self) -> Option<&DepositContract> {
+        self.inner.deposit_contract()
+    }
+
+    fn genesis_hash(&self) -> B256 {
+        self.inner.genesis_hash()
+    }
+
+    fn prune_delete_limit(&self) -> usize {
+        self.inner.prune_delete_limit()
+    }
+
+    fn display_hardforks(&self) -> Box<dyn Display> {
+        Box::new(ChainSpec::display_hardforks(self))
+    }
+
+    fn genesis_header(&self) -> &Header {
+        self.inner.genesis_header()
+    }
+
+    fn genesis(&self) -> &Genesis {
+        self.inner.genesis()
+    }
+
+    fn max_gas_limit(&self) -> u64 {
+        self.inner.max_gas_limit()
+    }
+
+    fn bootnodes(&self) -> Option<Vec<NodeRecord>> {
+        self.inner.bootnodes()
+    }
+}
+
+impl Hardforks for ScrollChainSpec {
+    fn fork<H: reth_chainspec::Hardfork>(&self, fork: H) -> reth_chainspec::ForkCondition {
+        self.inner.fork(fork)
+    }
+
+    fn forks_iter(
+        &self,
+    ) -> impl Iterator<Item = (&dyn reth_chainspec::Hardfork, reth_chainspec::ForkCondition)> {
+        self.inner.forks_iter()
+    }
+
+    fn fork_id(&self, head: &Head) -> ForkId {
+        self.inner.fork_id(head)
+    }
+
+    fn latest_fork_id(&self) -> ForkId {
+        self.inner.latest_fork_id()
+    }
+
+    fn fork_filter(&self, head: Head) -> ForkFilter {
+        self.inner.fork_filter(head)
+    }
+}
+
+impl EthereumHardforks for ScrollChainSpec {
+    fn get_final_paris_total_difficulty(&self) -> Option<U256> {
+        self.inner.get_final_paris_total_difficulty()
+    }
+
+    fn final_paris_total_difficulty(&self, block_number: u64) -> Option<U256> {
+        self.inner.final_paris_total_difficulty(block_number)
+    }
+}
+
+impl ScrollHardforks for ScrollChainSpec {}
+
+impl From<Genesis> for ScrollChainSpec {
+    fn from(genesis: Genesis) -> Self {
+        use reth_scroll_forks::ScrollHardfork;
+        let scroll_genesis_info = ScrollGenesisInfo::extract_from(&genesis);
+        let genesis_info = scroll_genesis_info.scroll_chain_info.genesis_info.unwrap_or_default();
```

Comment:
Why do we silently ignore error and fall back to default here?

Reply:
make sense. change to expect